Templates/Per Tenant Extension/.github/workflows/CICD.yaml

```diff
@@ -153,7 +176,7 @@ jobs:
       projectDependenciesJson: ${{ needs.Initialization.outputs.projectDependenciesJson }}
       secrets: 'licenseFileUrl,codeSignCertificateUrl,*codeSignCertificatePassword,keyVaultCertificateUrl,*keyVaultCertificatePassword,keyVaultClientId,gitHubPackagesContext,applicationInsightsConnectionString'
       publishThisBuildArtifacts: ${{ needs.Initialization.outputs.workflowDepth > 1 }}
-      publishArtifacts: ${{ github.ref_name == 'main' || startswith(github.ref_name, 'release/') || needs.Initialization.outputs.deliveryTargetsJson != '[]' || needs.Initialization.outputs.environmentCount > 0 }}
+      publishArtifacts: ${{ needs.Initialization.outputs.PublishArtifacts == 'True' }}
```

Comment:
> What's a scenario where you run CICD where you don't want to publish the artifacts?

When the CICD is run on a branch that's not _main_ or a release branch.

> Could publish artifacts just be true always for CICD?

Changing it to _true_ always might be breaking (haven't tested, though). If CICD is run on a dev branch, then the dev artifacts might potentially get deployed/delivered.

@freddydk any input here?

Reply:
> Changing it to true always might be breaking (haven't tested, though). If CICD is run on a dev branch, then the dev artifacts might potentially get deployed/delivered.

To be fair the D in CICD does stand for deploy/deliver 😄 If you don't want to deploy / deliver should you really be running the CIC**D** workflow? 

If you just want to test that your branch builds, and the tests pass shouldn't you be running a PR Build?

Reply:
Earlier, artifacts was published for main + release branches + if any delivery or deployment was true.
At the same time CICDPushBranches was also main and release - but many partners are changing this to run CI/CD on dev branches and they could even remove release branches if they decide to run CI/CD manually.
If they do this - then they cannot create hotfixes anymore after this change - or?

Reply:
Since the branch name is in the artifacts pushed - we could decide to push always - which would then remove the need for the thisbuild artifacts (I guess) - I need to think whether there are any unwanted consequences with this...

Reply:
CICD on dev branches sounds wild. 

I think the Deploy/Deliver part won't succeed at the moment as there will be no artifacts when CICD is run on a dev branch.
But if we change it to _always_ produce artifacts, then wouldn't Deploy/Deliver pick them up, potentially deploying/delivering artifacts from dev branches?

Reply:
Resorted to just add the `startswith(github.ref_name, 'releases/')` check. This shouldn't break existing scenarios.